app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt

```diff
@@ -155,35 +188,97 @@ class ConversationViewModel(
     )
 
     init {
-        viewModelScope.launch(Dispatchers.IO) {
+        viewModelScope.launch(Dispatchers.Default) {
             repository.recipientUpdateFlow(threadId)
                 .collect { recipient ->
-                    if (recipient == null && _uiState.value.conversationExists) {
-                        _uiState.update { it.copy(conversationExists = false) }
+                    _uiState.update {
+                        it.copy(
+                            shouldExit = recipient == null,
+                            showInput = shouldShowInput(recipient),
+                            enableInputMediaControls = shouldEnableInputMediaControls(recipient),
+                            messageRequestState = buildMessageRequestState(recipient),
+                        )
                     }
                 }
         }
+    }
 
-        // listen to community write access updates from this point
-        communityWriteAccessJob?.cancel()
-        communityWriteAccessJob = viewModelScope.launch {
-            OpenGroupManager.getCommunitiesWriteAccessFlow()
```

Comment:
Is there a reason for this logic disappearing completely? How do we get updates when the write access changes in a community?
There is a case where you can react with a specific emoji to gain admin access at which point the UI needs to update and display the input bar, and updating the underlying community object

Reply:
There's a consolidation happening in this file, if you look at `shouldShowInput`, it tries to put all the input hiding logic into one place. As the conversation view model **should** trigger the UI update whenever anything related to recipient has changed. Happy to test out if it still works for you said scenario, if not we should fix the reactive part

Reply:
Tested from the SOGS and the input access isn't dynamic anymore